.dotnet/src/Custom/Images/GeneratedImageSize.cs

```diff
@@ -0,0 +1,86 @@
+namespace OpenAI.Images;
+
+/// <summary>
+/// Represents the available output dimensions for generated images.
+/// </summary>
+public partial class GeneratedImageSize
+{
+    /// <summary>
+    /// Gets the desired width, in pixels, for an image.
+    /// </summary>
+    public int Width { get; }
+
+    /// <summary>
+    /// Gets the desired height, in pixels, for an image.
+    /// </summary>
+    public int Height { get; }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="GeneratedImageSize"/>.
+    /// </summary>
+    /// <remarks>
+    /// <b>Note:</b> arbitrary dimensions are not supported and a given model will only support a set of predefined
+    /// sizes. If supported dimensions are not known, try using one of the static properties like <see cref="Square"/>.
+    /// </remarks>
+    /// <param name="width"> The desired width, in pixels, for an image. </param>
+    /// <param name="height"> The desired height, in pixels, for an image. </param>
+    public GeneratedImageSize(int width, int height)
+    {
+        Width = width;
+        Height = height;
+    }
+
+    /// <summary>
+    /// A small, square image with 256 pixels of both width and height.
+    /// <para>
+    /// Supported <b>only</b> for the older <c>dall-e-2</c> model.
+    /// </para>
+    /// </summary>
+    public static readonly GeneratedImageSize TinySquare = new(256, 256);
```

Comment:
I have to say I am not a big fan of these names. I like change to make this a struct, but I think we should just keep the actual sizes in the names of the values.

Reply:
Fair; I agree with Jose in not liking the old names, either, so hopefully there's a better pattern. How does Jose's proposal to use labels in the form of `$"W{Width}xH{Height}`, like `W1792xH1024`, look to you? That gets rid of cutesy names, redundant names, and also disambiguates which dimension is which.

Reply:
I like the SizeHxW convention. I think it's customary to put horizontal dimensions first. But I would be ok with the $"W{Width}xH{Height} convention, if you guys prefer it.